.github/workflows/unit_test_matrix.yml

```diff
@@ -0,0 +1,76 @@
+name: Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  unit_tests:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        python_version: ${{ matrix.python_versions }}
+        tf_version: ${{ matrix.tf_versions }}
+        onnx_opset: ${{ matrix.onnx_opsets }}
+        include:
+          - python_versions: ['3.8']
+            tf_versions: ['2.9.0']
+            onnx_opsets: ['17', '16', '14']
+            platforms: ['linux', 'windows']
+          - python_versions: ['3.9']
+            tf_versions: ['2.15.0']
+            onnx_opsets: ['17', '16', '14']
+            platforms: ['linux', 'windows']
+          - python_versions: ['3.7']
+            tf_versions: ['1.15.5']
+            onnx_opsets: ['18', '17', '16', '15']
+            platforms: ['windows']
+          - python_versions: ['3.7']
+            tf_versions: ['1.15.5']
+            onnx_opsets: ['18', '17', '16']
+            platforms: ['linux']
+          - python_versions: ['3.10']
+            tf_versions: ['2.14.0']
+            platforms: ['linux', 'windows']
+          - python_versions: ['3.9']
+            tf_versions: ['2.12.0']
+            platforms: ['linux', 'windows']
+          - python_versions: ['3.8']
+            tf_versions: ['2.10.0', '2.11.0']
+            platforms: ['linux', 'windows']
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Run unit tests
+        run: pytest tests/
+
+      - name: Generate coverage report
+        if: ${{ matrix.report_coverage == 'True' }}
+        run: coverage report
```

Comment:
These two steps can be combined. See https://github.com/onnx/onnx/blob/main/.github/workflows/main.yml

`--cov-report=xml --cov-append --cov-branch --junit-xml pytest.xml`

Reply:
You would need https://pypi.org/project/pytest-cov/